lib/adb.js

```diff
@@ -554,13 +570,34 @@ ADB.prototype.installApp = function installApp(deviceId, apkFile, opts, callback
 };
 
 /**
+ * Returns the ps output of the specified app and device/emulator, if running.
+ * @param {String} deviceId - The id of the device or emulator
+ * @param {ADB~psCallback} callback - A callback that is fired once ps is executed
+ */
+ADB.prototype.ps = function ps(deviceId, callback) {
+	var output = function (err, data) {
+		if (err) {
+			callback(err);
+		} else {
+			// old ps, does not support '-A' parameter
+			if (data.toString().startsWith('bad pid \'-A\'')) {
+				this.shell(deviceId, 'ps', output);
+			} else {
+				callback(null, data);
+			}
+		}
+	}.bind(this);
+	this.shell(deviceId, 'ps -A', output);
```

Comment:
`ps -A` does nothing on my device.

```
chris@yojimbo (tmp):~/appc/node-titanium-sdk$ adb shell
shell@flo:/ $ ps -A
USER      PID   PPID  VSIZE  RSS   WCHAN            PC  NAME
shell@flo:/ $
shell@flo:/ $ ps
USER      PID   PPID  VSIZE  RSS   WCHAN            PC  NAME
root      1     0     2216   820   sys_epoll_ 00000000 S /init
root      2     0     0      0       kthreadd 00000000 S kthreadd
root      3     2     0      0     run_ksofti 00000000 S ksoftirqd/0
root      6     2     0      0     cpu_stoppe 00000000 S migration/0
root      16    2     0      0     rescuer_th 00000000 S khelper
root      17    2     0      0     rescuer_th 00000000 S suspend_sys_syn
root      18    2     0      0     rescuer_th 00000000 S suspend
root      23    2     0      0     irq_thread 00000000 S irq/203-msmdata
<snip>
```

I see you are checking if the output prints an error, but running `ps -A` on my device does not display an error. You basically need to check for the `bad pid '-A'` OR if it just returned the header. Maybe even run `ps` by default and fallback to `ps -A` if needbe.